File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/AlbumDetailsContract.java
package code.name.monkey.retromusic.mvp.contract;


import code.name.monkey.retromusic.model.Album;
import code.name.monkey.retromusic.mvp.BasePresenter;
import code.name.monkey.retromusic.mvp.BaseView;

public interface AlbumDetailsContract {

  interface AlbumDetailsView extends BaseView<Album> {

  }

  interface Presenter extends BasePresenter<AlbumDetailsView> {

    void loadAlbumSongs(int albumId);
  }
}

File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/AlbumDetailsContract.kt
package code.name.monkey.retromusic.mvp.contract


import code.name.monkey.retromusic.model.Album
import code.name.monkey.retromusic.mvp.BasePresenter
import code.name.monkey.retromusic.mvp.BaseView

interface AlbumDetailsContract {

    interface AlbumDetailsView : BaseView<Album>

    interface Presenter : BasePresenter<AlbumDetailsView> {

        fun loadAlbumSongs(albumId: Int)
    }
}
